component/prometheus/exporter/process/process.go

```diff
@@ -0,0 +1,79 @@
+package process
+
+import (
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/prometheus/exporter"
+	"github.com/grafana/agent/pkg/integrations"
+	"github.com/grafana/agent/pkg/integrations/process_exporter"
+	exporter_config "github.com/ncabatoff/process-exporter/config"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:    "prometheus.exporter.process",
+		Args:    Arguments{},
+		Exports: exporter.Exports{},
+		Build:   exporter.New(createIntegration, "process"),
+	})
+}
+
+func createIntegration(opts component.Options, args component.Arguments) (integrations.Integration, error) {
+	cfg := args.(Arguments)
+	return cfg.Convert().NewIntegration(opts.Logger)
+}
+
+// DefaultArguments holds the default arguments for the prometheus.exporter.process
+// component.
+var DefaultArguments = Arguments{
+	ProcFSPath: "/proc",
+	Children:   true,
+	Threads:    true,
+	SMaps:      true,
+	Recheck:    false,
+}
+
+// Arguments configures the prometheus.exporter.process component
+type Arguments struct {
+	ProcessExporter []MatcherGroup `river:"process_names,block,optional"`
```

Comment:
Typically blocks should have singular names, since it's weird to say 

```river 
rules {
  ... 
}

rules { 
  ...
}
```

vs 

```
rule { 

}

rule {

}
```

Is `process_names` right here? Or should it be given a singular name?

Reply:
This is kept over from the old config. Since `process_names` ultimately can actually match based on a combination of (1) the name in `/proc/<pid>/stat`, (2) the base name or path to the file, or (3) a regex of cmdline arguments, maybe something more generic and singular like `process_matcher` or `matcher` would be more appropriate?

So an example config would look like:
```
prometheus.exporter.process "example_process" {
    process_matcher {
        name = "non_flow_agents"
        comm = ["grafana-agent"]
        cmdline = ["*config.file*"]
    }

    process_matcher {
        name = "firefox"
        comm = ["firefox-bin"]
    }
}
```

Reply:
`matcher` seems best, I think process is probably implied given the component it's in.